File: testgenerator.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long
string test = "";

int genint(int a, int b, int k, bool add) {
  if (a == b) {
    if (add) test += to_string(a);
    return a;
  }
  int ans = (rand() % (b / k - a / k) + (a / k + (a % k != 0))) * k;
  if (add) test += to_string(ans);
  return ans;
}

string genstr(int a, int b, int k, vector <string> p, bool add) {
  int l = genint(a, b, k, 0);
  vector <char> symbols;
  for (int i = 0; i < p.size(); i++) {
    if (p[i].size() == 1) {
      symbols.push_back(p[i][0]);
    } else {
      for (char j = p[i][0]; j <= p[i][2]; j++) {
        symbols.push_back(j);
      }
    }
  }
  string ans = "";
  for (int i = 0; i < l; i++) {
    ans += symbols[genint(0, symbols.size(), 1, 0)];
  }
  if (add) test += ans;
  return ans;
}

signed main() {
  // generating test

  // ---------------
  freopen("test", "w", stdout);
  cout << test;
}

File: testgenerator.py
import random

test = ''

def genint(a, b, k, add):
    global test
    n = random.randint(a // k + int(a % k != 0), b // k) * k
    # n - random number (from a to b) divisible by k
    if add:
        test += str(n)
    return n

def genstr(a, b, k, p, add):
    global test
    l = random.randint(a // k + int(a % k != 0), b // k) * k
    # l - length (from a to b) divisible by k
    s = ''
    symbols = []
    for i in p:
        if len(i) == 1:
            symbols.append(i)
        else:
            for j in range(ord(i[0]), ord(i[2]) + 1):
                symbols.append(chr(j))
    # symbols - all possible symbols can use in string
    for i in range(l):
        s += symbols[random.randint(0, len(symbols) - 1)]
    if add:
        test += s
    return s

# generating test

# ---------------

test_file = open('test', 'w+')
test_file.write(test)
test_file.close()

# print('--test--\n\n', test, sep='')
